chat-plugins/info.js

```diff
@@ -1004,6 +1004,209 @@ var commands = exports.commands = {
+					if (Number.isInteger(Number(searchedWords[k]))) {
+						if (basePower) return this.sendReplyBox("Only specify a number for base power once.");
+						basePower = parseInt(searchedWords[k]);
+						if (basePower < 70 || 100 < basePower) return this.sendReplyBox('No berries inflict ' + basePower + 'bp damage when used with Natural Gift.');
+					}
+				}
+			}
+
+			for (var n in Tools.data.Items) {
+				var item = Tools.getItem(n);
+				if (!item.isBerry) continue;
+
+				if (basePower && type) {
+					if (item.naturalGift.basePower === basePower && item.naturalGift.type === type) foundItems.push(item.name);
+				} else if (basePower) {
+					if (item.naturalGift.basePower === basePower) foundItems.push(item.name);
+				} else {
+					if (item.naturalGift.type === type) foundItems.push(item.name);
+				}
+			}
+		} else {
+			var bestMatched = 0;
+			for (var n in Tools.data.Items) {
+				var item = Tools.getItem(n);
+				var matched = 0;
+				var descWords = item.desc.toLowerCase().replace('-', ' ').replace(/[^a-z0-9\s\/]/g, '').replace(/(\D)\./, function (p0, p1) { return p1; }).split(" ");
+
+				for (var k = 0; k < searchedWords.length; k++) {
+					if (descWords.indexOf(searchedWords[k]) >= 0) {
+						matched++;
+					}
+				}
+
+				if (matched >= bestMatched && matched >= (searchedWords.length * 3 / 5)) foundItems.push(item.name);
+				if (matched > bestMatched) bestMatched = matched;
+			}
+
+			// iterate over found items again to make sure they all are the best match
+			for (var l = 0; l < foundItems.length; l++) {
+				var item = Tools.getItem(foundItems[l]);
+				var matched = 0;
+				var descWords = item.desc.toLowerCase().replace('-', ' ').replace(/[^a-z0-9\s\/]/g, '').replace(/(\D)\./, function (p0, p1) { return p1; }).split(" ");
+
+				for (var k = 0; k < searchedWords.length; k++) {
+					if (descWords.indexOf(searchedWords[k]) >= 0) {
+						matched++;
+					}
+				}
+
+				if (matched !== bestMatched) {
+					foundItems.splice(l, 1);
+					l--;
+				}
+			}
+		}
+
+		var resultsStr = "";
+		if (foundItems.length > 0) {
+			if (showAll || foundItems.length <= output + 5) {
+				foundItems.sort();
+				resultsStr += foundItems.join(", ");
+			} else {
+				resultsStr += foundItems.slice(0, output).join(", ") + ", and " + string(foundItems.length - output) + " more. <font color=#999999>Redo the search with 'all' as a search parameter to show all results.</font>";
+			}
+		} else {
+			resultsStr = "No items found. Try a more general search";
+		}
+		return this.sendReplyBox(resultsStr);
+	},
+	itemsearchhelp: ["/itemsearch [move description] - finds items that match the given key words.",
+	"/itemsearch [fling] [base power] [effect] - finds items with the given fling behavior.",
+	"/itemsearch [natural gift] [base power] [type] - finds berries with the given natural gift behavior."],
```

Comment:
Yeah flags would definitely help make this command work cleaner, but I'd need to hear that's something @Zarel is okay with before trying to figure out the specifics of what to actually do with the flags. I personally don't even know if the better functionality is worth what it would take to add to the file (other than avoiding the embarrassment of having a command that feels sloppy). The only other use of having flags on items I could think of is potentially making it easier for OMs to target specific types of items to change their effects with a more general coding structure, but that's kind of a stretch.

Personally I wouldn't mind if this command only worked on berries because they're the most difficult sort of item to keep track of, but I could understand why others might see a command dedicated to searching berries just seemed obscure and overly niche.